test/TermCongr.lean

```diff
@@ -119,6 +119,20 @@ example (f g : Nat → Nat → Prop) (h : f = g) :
     (∀ x, ∃ y, f x y) ↔ (∀ x, ∃ y, g x y) :=
   congr(∀ _, ∃ _, $(by rw [h]))
 
+namespace SubsingletonDependence
+/-!
+The congruence theorem generator had a bug that leaked fvars.
+Reported at https://leanprover.zulipchat.com/#narrow/stream/113489-new-members/topic/what.20the.20heq.2C.20PosSemidef/near/434202080
+-/
+
+def f {α : Type} [DecidableEq α] (n : α) (_ : n = n) : α := n
+
+lemma test (n n' : Nat) (h : n = n') (hn : n = n) (hn' : n' = n') :
+    f n hn = f n' hn' := by
+  exact congr(f $h ‹_›)
```

Comment:
Can you add a test with `_` too? Even if it errors, I think it's useful to check it errors in the right way.

Reply:
I'm not sure I understand what that would be testing -- the but turns out not to have to do with `‹_›`. (It does work with `_` in place of `‹_›`, but that's because this argument can be solved for by unification with the goal.)

Reply:
My thought is simply that this also fails on master;
```
import Mathlib

def f {α : Type} [DecidableEq α] (n : α) (_ : n = n) : α := n

lemma test (n n' : Nat) (h : n = n') (hn : n = n) (hn' : n' = n') :
    f n hn = f n' hn' := by
  exact congr(f $h _)
```
and testing both `_` and `‹_›` makes it clear that it is not a quirk unique to `‹_›`.

Reply:
They both fail for the same reason, and I think it's important to follow the principle to test specific features.

That said, I added a `_` version, just to see that unification using the expected type actually works in this context. I'm still not convinced about testing this here, but I managed to work both versions into the same example.

Reply:
Thanks!